middleware/kubernetes/k8sclient/dataobjects.go

```diff
@@ -5,9 +5,16 @@ import (
     "net/http"
 )
 
+// getK8sAPIResponse wraps the http.Get(url) function to provide dependency
+// injection for unit testing.
+var getK8sAPIResponse = func(url string) (resp *http.Response, err error) {
+	resp, err = http.Get(url)
+	return resp, err
+}
+
 
-func getJson(url string, target interface{}) error {
-    r, err := http.Get(url)
+func parseJson(url string, target interface{}) error {
```

Comment:
target is used in the call to json.NewDecoder(r.Body).Decode(target) on line 23

Not sure what extra level of indirection you are asking about. Assuming you are asking about the "getK8sAPIResponse" function pointer. Using the variable function pointer allows me to do dependency injection in the unit tests by effectively overriding http.Get() with a custom function for testing.